pwa-shell/src/ui/menu/LogoutContainer.js

```diff
@@ -17,19 +17,20 @@ export const mapStateToProps = state => ({
   userFullname: getUserFullname(state),
 });
 
-export const mapDispatchToProps = (dispatch, { useKeycloak, auth }) => ({
-  loadUserProfile: useKeycloak
+export const mapDispatchToProps = (dispatch, { auth }) => {
+  const logout = get(auth, 'logout');
+  const loadUserProfile = logout
```

Comment:
why the condition is the existence of `logout`?

Reply:
logout on auth object does not exist on keycloak-type auth object. or do you prefer to place a property on auth object to determine the auth type, like, for example, `auth.type`?

Reply:
Beware, the KC object has a logout method!
I'd use an object with methods for default auth as well, so you don't have to make any check. Just call logout.

Reply:
Hi @gidesan, thanks for your opinion, I have use lodash's method 'get' instead to find auth's logout method otherwise results to a blank function. Also for the fetchprofile I can't find a way to detect its auth type but I instead use similar to what I use on logout. I just find 'realm' property in the auth object (which only exist in keycloak auth).

Reply:
IMO it's a weak condition because it involves some internal details that could change. I think we have two options:
1) converge to a single interface and add a load user profile method for both auth objects (default and kc)
2) use the `useKeycloak` condition but consider it as a technical debt that could be removed if we add OIDC to the default entando auth

Reply:
I took a step back and thought of considering your #1 option but in a different way - I moved the fetch profile for default auth from LoginContainer to a new container for DefaultAuthProvider. There is a technical debt for me which I did add a throttle before invoking fetch profile. there's a bit of an issue on `store` being null at the start of the app when calling fetch. Maybe it is too early on calling fetch profile hence added the throttle call. But aside from codewise, this is working well already. Let me know your thoughts on this.

Reply:
I'm afraid that `throttle` won't always be reliable. @nicpuddu what do you think about that?
Anyway, let's consider this technical debt should be paid very soon. My plan is:

- extract auth flow from the PWA to an external package (we could put in `frontend-libraries` repo)
- ensure the new auth package is decoupled from `apiManager`
- use the new auth package in PWA, appbuilder and potentially SME projects

Reply:
sounds good @gidesan. Let's do that then

Reply:
__throttle__ is meant to be there for testing really, I don't like the idea of it being used in actual production code.

Reply:
okay. I'll find a way that I won't be calling that method. What I'll solve is `fetchUserProfile` should not be called before apiManager's `config` method has been called. I'll now definitely go to the direction on making load Profile method for both kinds of auth object provided from withAuth.

Reply:
@gidesan @nicpuddu no more throttle method and used same load user method on both auth objects. Please review